pkg/services/notary/notary.go

```diff
@@ -316,7 +317,28 @@ func (n *Notary) finalize(tx *transaction.Transaction) error {
 			break
 		}
 	}
-	return n.onTransaction(tx)
+	newTx, err := updateTxSize(tx)
+	if err != nil {
+		return fmt.Errorf("failed to update completed transaction's size: %w", err)
+	}
+
+	return n.onTransaction(newTx)
+}
+
+// updateTxSize returns transaction with re-calculated size and an error.
+func updateTxSize(tx *transaction.Transaction) (*transaction.Transaction, error) {
+	bw := io.NewBufBinWriter()
+	tx.EncodeBinary(bw.BinWriter)
```

Comment:
`tx.Bytes()`?

Reply:
But what about encoding error if it occurs?

Reply:
You can still check for `nil` and return error, although it won't be specific, so current code is somewhat better here.

Reply:
Although it's not possible. Fixed.

Reply:
Reverted.